luaui/Widgets/gui_unit_group_number.lua

```diff
@@ -110,24 +110,25 @@ local function initGL4()
 	return true
 end
 
-local function RemovePrimitive(unitID)
-	if unitGroupVBO.instanceIDtoIndex[unitID] then
-		local oldgroup = unitIDtoGroup[unitID]
-		grouptounitID[oldgroup][unitID] = nil
+local function RemovePrimitive(unitID, groupID)
+	groupID = groupID or unitIDtoGroup[unitID]
+
+	if groupID then
+		grouptounitID[groupID][unitID] = nil
 		unitIDtoGroup[unitID] = nil
-		popElementInstance(unitGroupVBO, unitID)
 	end
-end
 
-function widget:VisibleUnitRemoved(unitID) -- E.g. when a unit dies
```

Comment:
I have no idea what's going on with `VisibleUnitsRemoved` but its getting triggered when a unit is added to the game for me.